File: game_server/ServerProtocol/src/Common/GuildDungeonEnum/EGuildDungeon_MonsterType.java
package Common.GuildDungeonEnum;

/*********
 * 联盟副本怪物类型
 **/
public enum EGuildDungeon_MonsterType {
	NONE, //0 ==== 
	COMMON, //1 ==== 普通怪物
	SENIOR, //2 ==== 高级怪物
	BOSS, //3 ==== boss
	;
public static final EGuildDungeon_MonsterType[]  EGuildDungeon_MonsterType_Values = EGuildDungeon_MonsterType.values();
public static final int EGuildDungeon_MonsterType_Length = EGuildDungeon_MonsterType_Values.length;
public static EGuildDungeon_MonsterType EGuildDungeon_MonsterType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EGuildDungeon_MonsterType_Length){ return null; }
	return EGuildDungeon_MonsterType_Values[_ivalue];
}
}

File: game_client/Assets/Scripts/ClientProtocol/Common/GuildDungeonEnum/EGuildDungeon_MonsterType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.GuildDungeonEnum
{

/// <summary>
/// 联盟副本怪物类型
/// </summary>
public enum EGuildDungeon_MonsterType {
	NONE, //0 ==== 
	[InspectorName("COMMON - idx[1] - 普通怪物")]
	COMMON, //1 ==== 普通怪物
	[InspectorName("SENIOR - idx[2] - 高级怪物")]
	SENIOR, //2 ==== 高级怪物
	[InspectorName("BOSS - idx[3] - boss")]
	BOSS, //3 ==== boss
}

public class EGuildDungeon_MonsterTypeComparer : IEqualityComparer<EGuildDungeon_MonsterType>{
	public bool Equals(EGuildDungeon_MonsterType x, EGuildDungeon_MonsterType y) { return x == y; }
	public int GetHashCode(EGuildDungeon_MonsterType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 4;
}
}

